src/module/account/settings/setting-form.component.test.tsx

```diff
@@ -0,0 +1,57 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import SettingForm from './setting-form.component';
+
+function renderSettingPage() {
+  render(<SettingForm />);
+}
+
+describe('setting', () => {
+  it('should show a required field warning for each empty input field', async () => {
+    renderSettingPage();
+    const submitButton = screen.getByRole('button', { name: /submit/i });
+    await userEvent.click(submitButton);
+    expect(await screen.findByText(/display name is required/i)).toBeVisible();
+    expect(await screen.findByText(/email is required/i)).toBeVisible();
+    expect(await screen.findByText(/password is required/i)).toBeVisible();
+  });
+
+  it('should show invalid field errors for each invalid input field', async () => {
+    renderSettingPage();
+    const displayNameInput = screen.getByPlaceholderText(/display name/i);
+    const emailInput = screen.getByPlaceholderText(/email/i);
+    const passwordInput = screen.getByPlaceholderText(/password/i);
+
+    await userEvent.type(displayNameInput, 'a');
+    await userEvent.type(emailInput, 'a');
+    await userEvent.type(passwordInput, 'a');
+
+    const submitButton = screen.getByRole('button', { name: /submit/i });
+    await userEvent.click(submitButton);
+
+    expect(await screen.findByText(/at least 4/i)).toBeVisible();
+    expect(await screen.findByText(/Invalid email address/i)).toBeVisible();
+    expect(await screen.findByText(/at least 8/i)).toBeVisible();
+
+    await userEvent.type(displayNameInput, 'asd!');
+    await userEvent.type(passwordInput, 'aaaaaaa2');
+    expect(
+      await screen.findByText(
+        /Display name must contain only letters, numbers, hyphens and underscores/i
+      )
+    ).toBeVisible();
+    expect(
+      await screen.findByText(
+        /Password must contain at least one letter, one number and one special character/i
+      )
+    ).toBeVisible();
+
+    await userEvent.type(displayNameInput, 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa');
```

Comment:
we tried to mock the input data as close to the as real world case as possible, please try to use something looks better

Reply:
change the mocked password more close to real word case